### DISCUSSION TITLE
HttpUrl and AnyHttpUrl should have defaults for port when no port is specified

### DISCUSSION THREAD

**Discussion by davidolrik** [pydantic/pydantic#2450] [Feature Request]
When creating a `HttpUrl` or  `AnyHttpUrl`, the url scheme is required which should dictate a port if you don't specify one.
As it is now no port specified means `port` is `None`.

I propose that we look at the url scheme, and return `80` for  `http` or `443` for `https` if `port` is not specified.

**Comment by samuelcolvin** [comment]
I haven't looked in detail, but I think coincidentally #2447 might make this possible.

My main concern is that converting an `HttpUrl` back to a string should result in the same string as the original.

If we always populate the port, we might end up with

    'https://example.com/foobar' -> HttpUrl -> 'https://example.com:443/foobar'

which would be annoying

**Reply by davidolrik** [reply]
I wouldn't populate the port, just make the getter return the correct port when no port i specified.

**Reply by MihanixA** [reply]
Perhaps we could add `hidden_fields` member to hide a port field in `HttpUrl` for example?

**Reply by MihanixA** [reply]
Or `display_port: bool` to be specific

**Reply by MihanixA** [reply]
Committed to https://github.com/samuelcolvin/pydantic/pull/2447
